### ISSUE TITLE
Adding unique constrains change the generation into OneToOne

### ISSUE THREAD

**Issue by nofalx** [Kononnable/typeorm-model-generator#135]
DB [postgres]
TypeORM version: 0.2.4

Python Django Code that is used to generate migration to the DB, the only difference i made that generated the bug is adding `unique_together` at the bottom of this code which will add unique constrain on DB level. The problem is generation behavior changed relationship too.
```
class TravelPlan(models.Model):
    id = models.UUIDField(primary_key=True, default=uuid.uuid4, editable=False)
    company = models.ForeignKey( Company, on_delete=models.CASCADE, null=True)
    name = models.CharField( max_length=128)
    extras = models.ManyToManyField( 'Extra', through='TravelPlanExtra')

    def __str__(self):
        return self.name

class Extra(models.Model):
    id = models.UUIDField(primary_key=True, default=uuid.uuid4, editable=False)
    title = models.CharField( max_length=256)
    view_position = models.SmallIntegerField( default=2, )

    def __str__(self):
        return self.title

class TravelPlanExtra(models.Model):
    id = models.UUIDField(primary_key=True, default=uuid.uuid4, editable=False)
    travel_plan = models.ForeignKey(TravelPlan,  on_delete=models.CASCADE, related_name='TravelPlanOption')
    extra = models.ForeignKey(Extra, on_delete=models.CASCADE, related_name='TravelPlanOption')
    details = models.CharField( max_length=256, blank=True)
    addidtional_charges = models.IntegerField()

    def __str__(self):
        return self.travel_plan.name + ', ' + self.extra.title

    class Meta:
        unique_together = ('travel_plan', 'extra',)
```

---
### Before Unique Constrain
Before adding the unique together to the DB migrations output of the generator was:
```
...
    @ManyToOne(type=>travel_extra, travel_extra=>travel_extra.travel_travelplanextras,{  nullable:false, })
    @JoinColumn({ name:'extra_id'})
    extra_:travel_extra | null;


   
    @ManyToOne(type=>travel_travelplan, travel_travelplan=>travel_travelplan.travel_travelplanextras,{  nullable:false, })
    @JoinColumn({ name:'travel_plan_id'})
    travel_plan_:travel_travelplan | null;
...
```
---
### After Unique Constrain
But after adding the unique constrain on `extra_` and `travel_plan_`  on Database level this line was added
```
@Index("travel_travelplanextra_travel_plan_id_extra_id_f825ca51_uniq",["extra_","travel_plan_",],{unique:true})
```
And the relationship changed:
```
...
    @OneToOne(type=>travel_extra, travel_extra=>travel_extra.travel_travelplanextra,{  nullable:false, })
    @JoinColumn({ name:'extra_id'})
    extra_:travel_extra | null;


   
    @OneToOne(type=>travel_travelplan, travel_travelplan=>travel_travelplan.travel_travelplanextra,{  nullable:false, })
    @JoinColumn({ name:'travel_plan_id'})
    travel_plan_:travel_travelplan | null;
...
```

**Comment by nofalx** [comment]
correction for typeorm version is `0.2.24`

**Comment by Kononnable** [comment]
Fixed and published as `typeorm-model-generator@0.2.25`.

**Comment by nofalx** [comment]
Just wanna say thanks for the fast fix. Tested it and its fixed. This is a very amazing useful project.
